src/main/java/org/opensearch/timeseries/model/Config.java

```diff
@@ -0,0 +1,550 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.timeseries.model;
+
+import static org.opensearch.timeseries.constant.CommonMessages.INVALID_CHAR_IN_RESULT_INDEX_NAME;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.util.Strings;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.timeseries.annotation.Generated;
+import org.opensearch.timeseries.constant.CommonMessages;
+import org.opensearch.timeseries.constant.CommonName;
+import org.opensearch.timeseries.dataprocessor.FixedValueImputer;
+import org.opensearch.timeseries.dataprocessor.ImputationMethod;
+import org.opensearch.timeseries.dataprocessor.ImputationOption;
+import org.opensearch.timeseries.dataprocessor.Imputer;
+import org.opensearch.timeseries.dataprocessor.LinearUniformImputer;
+import org.opensearch.timeseries.dataprocessor.PreviousValueImputer;
+import org.opensearch.timeseries.dataprocessor.ZeroImputer;
+import org.opensearch.timeseries.settings.TimeSeriesSettings;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.ImmutableList;
+
+public abstract class Config implements Writeable, ToXContentObject {
```

Comment:
nit: should we maybe call this TimeSeries and then Forecaster and AD are extending this as they are a TimeSeries

Reply:
It is the configuration of time series analysis so I called it config. We also have time series analysis's job, checkpoint, etc.